IdentityCore/src/cache/crypto/MSIDAssymetricKeyKeychainGenerator.m

```diff
@@ -295,9 +299,14 @@ - (BOOL)logAndFillError:(NSString *)errorTitle status:(OSStatus)status error:(NS
     NSString *description = [NSString stringWithFormat:@"Operation failed with title \"%@\", status %ld", errorTitle, (long)status];
     MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"%@", description);
     
-    if (error)
-    {
-        *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, description, nil, nil, nil, nil, nil, NO);
+    if (error) {
+        NSError *underlyingError = nil;
+        if (status != NoStatus) {
+            // Underlying error is used to surface a system error code that caused the failure, if any.
+            underlyingError = MSIDCreateError(MSIDErrorDomain, (NSInteger)status, description, nil, nil, nil, nil, nil, NO);
```

Comment:
I think we should use MSIDKeychainErrorDomain instead of MSIDErrorDomain here

Reply:
Potentially. I'll check the other uses.

Reply:
I was wondering - why shouldn't we make the top-level error MSIDKeychainErrorDomain is the status is not kNoStatus ? In this case, we don't need the nested error. The status can directly go to the error code, and the problem is solved.  
i.e., if (status == kNoStatus) {
     return {MSIDErrorDomain , MSIDErrorInternal};
} else {
     return {MSIDKeychainErrorDomain , status};
}

Reply:
Spoke offline, agreed.